source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,39 @@
 class UrlsController < ApplicationController
+  def create
+    @url = Url.new(urls_params)
+
+    if @url.save
+      redirect_to urls_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @urls = Url.all
+  end
+
+  def new
+    @url = Url.new
+  end
+
+  def route_me
+    url = Url.find_by_short_url params[:short_url]
+
+    if url.nil?
+      redirect_to root_path
+    else
+      url.update('click_count' => url.click_count + 1)
```

Comment:
ActiveRecord [has a method for this kind of thing](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-increment).

There's a subtle bug as currently written. Do you see it?